pkg/monitor/monitorapi/construction.go

```diff
@@ -294,24 +295,37 @@ func (b *LocatorBuilder) withServer(serverName string) *LocatorBuilder {
 	return b
 }
 
+// TODO decide whether we want to allow "random" locator keys.  deads2k is -1 on random locator keys and thinks we should enumerate every possible key we special case.
 func (b *LocatorBuilder) KubeEvent(event *corev1.Event) Locator {
-	b.targetType = LocatorTypeKubeEvent
 
-	// WARNING: we're trying to use an enum for the locator keys, but we cannot know
-	// all kinds in a cluster. Instead we'll split the kind and name into two different Keys
-	// for Events:
-	b.annotations[LocatorKindKey] = event.InvolvedObject.Kind
-	b.annotations[LocatorNameKey] = event.InvolvedObject.Name
+	// When Kube Events are displayed, we need repeats of the same event to appear on one line. To do this
+	// we hash the event message, get the first ten characters, and add it as a hmsg locator key.
+	hash := sha256.Sum256([]byte(event.Message))
+	hashStr := fmt.Sprintf("%x", hash)[:10]
+	b.annotations[LocatorHmsgKey] = hashStr
 
-	if len(event.InvolvedObject.Namespace) > 0 {
-		b.annotations[LocatorNamespaceKey] = event.InvolvedObject.Namespace
+	if event.InvolvedObject.Kind == "Namespace" {
+		// namespace better match the event itself.
+		return b.
+			withNamespace(event.InvolvedObject.Name).
+			Build()
+	} else if event.InvolvedObject.Kind == "Node" {
+		return b.
+			withTargetType(LocatorTypeNode).
+			withNode(event.InvolvedObject.Kind).
```

Comment:
Are you wanting to set "Node" here or the actual node?

Reply:
Ooops!